app/browser/reducers/downloadsReducer.js

```diff
@@ -14,17 +14,25 @@ const {cancelDownload, pauseDownload, resumeDownload} = require('../electronDown
 const {CANCEL, PAUSE, RESUME} = require('../../common/constants/electronDownloadItemActions')
 const appActions = require('../../../js/actions/appActions')
 const userPrefs = require('../../../js/state/userPrefs')
+const getSetting = require('../../../js/settings').getSetting
 
 const downloadsReducer = (state, action) => {
   const download = action.downloadId ? state.getIn(['downloads', action.downloadId]) : undefined
   if (!download &&
       ![appConstants.APP_MERGE_DOWNLOAD_DETAIL,
         appConstants.APP_CLEAR_COMPLETED_DOWNLOADS,
         appConstants.APP_SELECT_DEFAULT_DOWNLOAD_PATH,
-        appConstants.APP_CHANGE_SETTING].includes(action.actionType)) {
+        appConstants.APP_CHANGE_SETTING,
+        appConstants.APP_SET_STATE].includes(action.actionType)) {
     return state
   }
   switch (action.actionType) {
+    case appConstants.APP_SET_STATE:
+      if (getSetting(settings.DOWNLOAD_DEFAULT_PATH) === '') {
+        const defaultPath = app.getPath('downloads')
```

Comment:
Where does electron/Muon default `downloads` to for each platform?

Reply:
It will try to get path of `chrome::DIR_DEFAULT_DOWNLOADS` in `atom_api_app.cc`
eventually leads to 
```c++
    case chrome::DIR_DEFAULT_DOWNLOADS:
#if defined(OS_ANDROID)
      if (!base::android::GetDownloadsDirectory(&cur))
        return false;
#else
      if (!GetUserDownloadsDirectory(&cur))
        return false;
      // Do not create the download directory here, we have done it twice now
      // and annoyed a lot of users.
#endif
      break;
```
in `chrome/common/chrome_paths.cc`
`GetUserDownloadsDirectory` has different implementation on different platforms
It will be `~/Downloads` as default